### DISCUSSION TITLE
Stores, Reducers and Effect Logic

### DISCUSSION THREAD

**Discussion by realisable** [mrpmorris/Fluxor#438] [Q&A] [answered]
Hello - I have a general question surrounding stores, reducers and effects (and I understand these are going to be in the line of 'how long is a piece of string')..

I have a particular complex UI which logically has a single store....

1. What should be a typically number of properties on a store?
2. Should reducers be short? Is it this normal to have an action with decently complex logic (20-30 lines) to reduce a state?
3. Similarly, should effects be equally short?

I ask as the examples are simple and I cannot find too many/any examples of a complex Flux pattern.

**Comment by mrpmorris** [comment, marked as answer]
The number of properties is irrelevant, what matters is whether or not they all belong together, only you can answer that.

As for number of lines of code.  They should do what they need to do, and no more. If that means 1000 lines then it does; however, you should consider splitting them up into separate methods at least - you might want to consider having code that you can reuse (services for effects, static methods for reducers).

**Reply by realisable** [reply]
Thanks for the quick response - it's kind of what I thought (standard engineering...don't do dumb stuff) - the question was to ensure we are going down the path correctly with the flux pattern before rolling out to multiple elements of our new UI.
